packages/features/troubleshooter/components/LargeCalendar.tsx

```diff
@@ -0,0 +1,140 @@
+  const event = useTroubleshooterStore((state) => state.event);
+  const calendarToColorMap = useTroubleshooterStore((state) => state.calendarToColorMap);
+  const { data: session } = useSession();
+  const startDate = selectedDate ? dayjs(selectedDate) : dayjs();
+
+  const { data: busyEvents } = trpc.viewer.availability.user.useQuery(
+    {
+      username: session?.user?.username || "",
+      dateFrom: startDate.startOf("day").utc().format(),
+      dateTo: startDate
+        .endOf("day")
+        .add(extraDays - 1, "day")
+        .utc()
+        .format(),
+      withSource: true,
+    },
+    {
+      enabled: !!session?.user?.username,
+    }
+  );
+
+  const { data: schedule } = useSchedule({
+    username: session?.user.username || "",
+    eventSlug: event?.slug,
+    eventId: event?.id,
+    timezone,
+    month: startDate.format("YYYY-MM"),
+  });
+
+  const endDate = dayjs(startDate)
+    .add(extraDays - 1, "day")
+    .toDate();
+
+  const availableSlots = useMemo(() => {
+    const availableTimeslots: CalendarAvailableTimeslots = {};
+    if (!schedule) return availableTimeslots;
+    if (!schedule?.slots) return availableTimeslots;
+
+    for (const day in schedule.slots) {
+      availableTimeslots[day] = schedule.slots[day].map((slot) => ({
+        start: dayjs(slot.time).toDate(),
+        end: dayjs(slot.time)
+          .add(event?.duration ?? 30, "minutes")
+          .toDate(),
+      }));
+    }
+
+    return availableTimeslots;
+  }, [schedule, event]);
+
+  const events = useMemo(() => {
+    if (!busyEvents?.busy) return [];
+
+    // TODO: Add buffer times in here as well just requires a bit of logic for fetching event type and then adding buffer time
+    //   start: dayjs(startTime)
+    //   .subtract((eventType?.beforeEventBuffer || 0) + (afterEventBuffer || 0), "minute")
+    //   .toDate(),
+    // end: dayjs(endTime)
+    //   .add((eventType?.afterEventBuffer || 0) + (beforeEventBuffer || 0), "minute")
+    //   .toDate(),
+
+    const calendarEvents = busyEvents?.busy.map((event, idx) => {
+      return {
+        id: idx,
+        title: event.title ?? `Busy`,
+        start: new Date(event.start),
+        end: new Date(event.end),
+        options: {
+          borderColor:
+            event.source && calendarToColorMap[event.source] ? calendarToColorMap[event.source] : "black",
+          status: BookingStatus.ACCEPTED,
+        },
+      };
+    });
+
+    if (busyEvents.dateOverrides) {
+      busyEvents.dateOverrides.forEach((dateOverride) => {
+        const dateOverrideStart = dayjs(dateOverride.start);
+        const dateOverrideEnd = dayjs(dateOverride.end);
+
+        if (!dateOverrideStart.isSame(dateOverrideEnd)) {
+          return;
+        }
+
+        const dayOfWeekNum = dateOverrideStart.day();
+
+        const workingHoursForDay = busyEvents.workingHours.find((workingHours) =>
+          workingHours.days.includes(dayOfWeekNum)
+        );
```

Comment:
Find working hours so we can use the start and end to make the correct "size" of busy time block on the calendar 

as busy times is just a start/end of midnight